### ISSUE TITLE
Optionally skip warnings of color-no-hex:true for custom-properties

### ISSUE THREAD

**Issue by ezhlobo** [stylelint/stylelint#2324]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

New option. I want to ignore rule `color-no-hex` for custom-properties.

> Which rule, if any, is this issue related to?

`color-no-hex`

> What CSS is needed to reproduce this issue?

```css
:root {
  --color: #000;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "color-no-hex": true
  }
}
```

> Which version of stylelint are you using?

`7.6.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`stylelint 'src/**/*.css' --config .stylelint.yml`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

I believe custom properties is a standard syntax now.

> What did you expect to happen?

I want to configure this rule to skip the warning if I use colors as a value of custom-property.

> What actually happened (e.g. what warnings or errors you are getting)?

It consideres warning in console.

I found related issue (https://github.com/stylelint/stylelint/issues/1006) but you suggested to create a new issue if it's related to custom-properties.

**Comment by m-allanson** [comment]
Taking reference from [`color-named`](https://github.com/stylelint/stylelint/tree/master/lib/rules/color-named) this could be added as a secondary option, either `ignore: ["custom-properties"]` or `ignoreProperties: ["/regex/", "string"]`.

**Comment by jeddy3** [comment]
> I want to ignore rule color-no-hex for custom-properties.

@ezhlobo Can you please tell us a bit more about what you're trying to achieve i.e. why you want to do this? For example, are you trying to ensure CSS variables are used for `<color>` values in the likes of `color` and `background-color`, or is it for another reason?

> this could be added as a secondary option, either `ignore: ["custom-properties"]` or `ignoreProperties: ["/regex/", "string"]`

I think adding such an option would involve rewriting the rule to use `postcss-value-parser` and so I think we need to be sure that this particular use case is being addressed in the most appropriate way.

**Comment by jeddy3** [comment]
Closing due to inactivity.
